x-pack/platform/plugins/private/upgrade_assistant/server/lib/data_streams/data_stream_migration_service.ts

```diff
@@ -321,5 +328,16 @@ export const dataStreamReindexServiceFactory = ({
         );
       }
     },
+
+    async readonlyIndices(indices: string[]) {
+      for (const index of indices) {
+        const unfreeze = await esClient.indices.unfreeze({ index });
+        const addBlock = await esClient.indices.addBlock({ index, block: 'read_only' });
```

Comment:
```suggestion
        const addBlock = await esClient.indices.addBlock({ index, block: 'write' });
```

Based on feedback from ES I think we need to use `write` here... But I may have outdated info

Reply:
We need to check I took this code from @gsoldevila so we'd need to update it on both ends. Ultimately we'd share this function call between data stream service and normal reindexing service